lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1949,24 +1949,17 @@ protected function get_css_variables( $nodes, $origins ) {
 	 * creates the corresponding ruleset.
 	 *
 	 * @since 5.8.0
+	 * @deprecated 7.0.0 Use WP_Style_Engine_Gutenberg::compile_css() instead.
 	 *
 	 * @param string $selector     CSS selector.
 	 * @param array  $declarations List of declarations.
 	 * @return string The resulting CSS ruleset.
 	 */
 	protected static function to_ruleset( $selector, $declarations ) {
-		if ( empty( $declarations ) ) {
-			return '';
-		}
+		_deprecated_function( __METHOD__, 'Gutenberg 7.0.0', 'WP_Style_Engine_Gutenberg::compile_css()' );
```

Comment:
In Core, this will be `WP_Style_Engine::compile_css()`

Reply:
This is a protected method, does it need a `_deprecated_function` call, or can it just be removed? I don't see any calls to it in `wordpress-develop` other than internal calls within the Theme JSON class.

Or is it to protect classes (like this one in Gutenberg) that might be extending the Theme JSON class? If so, then yes, deprecating sounds safer! (Did I just answer my own question? 😆)

Reply:
It's a good question. I'd actually be happy to delete it.

But I did it for the reason you mention (in case it's been extended) and this is despite the 4yo notice at the top of the class that says

```php
/**
 * Class that encapsulates the processing of structures that adhere to the theme.json spec.
 *
 * This class is for internal core usage and is not supposed to be used by extenders (plugins and/or themes).
 * This is a low-level API that may need to do breaking changes. Please,
 * use get_global_settings, get_global_styles, and get_global_stylesheet instead.
 *
 * @access private
 */
```

So do we care about folks who extended anyway? 🤔